### PR TITLE
docs: update Icons documentation and add examples

### PR DESCRIPTION
**Pull request by DiegoCardoso**
Update the Icons documentation page to move the current list to a new tab and use the main page to describe the new features added as part of the 6WP.

### CODE REVIEW

**Review by rolfsmeds — CHANGES_REQUESTED** [review]

File: articles/components/icons/index.asciidoc [hunk 1]
```diff
@@ -8,129 +10,464 @@ page-links:
...
-import com.vaadin.flow.component.icon.VaadinIcon;
-import com.vaadin.flow.router.Route;
 
-@Route("icons")
-public class IconsExample extends Div {
-  public MyApp() {
-    // tag::snippet[]
-    Icon lumoIcon = new Icon("lumo", "photo");
-    Icon vaadinIcon = new Icon(VaadinIcon.PHONE);
+.Example using FontAwesome
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply articles/components/icons/index.asciidoc]: Done.

File: articles/components/icons/index.asciidoc [hunk 2]
```diff
@@ -8,129 +10,464 @@ page-links:
...
+[source,tsx]
+----
+include::{root}/frontend/demo/component/icons/react/lumo-icons.tsx[render,tags=snippet,indent=0,group=React]
+----
+endif::[]
+--
+
+Lumo Icons are rendered on a 24×24 pixel canvas with a 16×16 pixel active area for the icon itself and 4 pixels of whitespace around the icon.
+
+The Default Icons tab contains a listing of the full Lumo Icons collection.
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply articles/components/icons/index.asciidoc]: Done.

File: articles/components/icons/index.asciidoc [hunk 3]
```diff
@@ -8,129 +10,464 @@ page-links:
...
+== Creating Custom Icon Collection APIs
+
+Within the application context, various icon libraries, including FontAwesome, Material Symbol, and others, may be utilized.  Implementing custom icon libraries, as exemplified by the `VaadinIcon` enumeration, simplifies developer workflows by eliminating the need to recall specific URL paths or remember which CSS classes to apply when working with font icons:
+
+[.example]
+--
+
+[source, java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/icons/FontAwesomeIcons.java[group=Java]
```
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply articles/components/icons/index.asciidoc]: Added an usage example:
image

**Review by DiegoCardoso — COMMENTED** [review]
Addressed comments on the main doc page. Gonna address the changes in the default icons page now.

**Review by tomivirkki — APPROVED** [review]

File: articles/components/icons/index.asciidoc [hunk 4]
```diff
@@ -8,129 +10,464 @@ page-links:
...
+----
+<Icon icon="vaadin:star" role="img" aria-label="Favorite" />
+----
+endif::[]
 
-For *PNG icons*, use the `<img>`/<<{articles}/application/resources/#the-image-component,`Image`>> component.
+--
 
 
 [discussion-id]`47B97C93-9646-4D2A-882F-C4F709D3D099`
```
Comment (DiegoCardoso) [inline articles/components/icons/index.asciidoc:473]: BTW, this has the same id as the `defaul-icons` tab. Should there be another discussion id for both? Which tab should keep the currend it?

**Review by tomivirkki — APPROVED** [review]

File: articles/components/icons/index.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (jouni) [reply articles/components/icons/index.asciidoc]: Unnecessary example group (Java).

**Review by rolfsmeds — APPROVED** [review]

File: articles/components/icons/index.asciidoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (DiegoCardoso) [reply articles/components/icons/index.asciidoc]: Removed.

**Pull request merged**